src/Pangraph/Internal/HexmlExtra.hs

```diff
@@ -5,24 +5,51 @@ import Data.ByteString (ByteString)
 import Text.XML.Hexml
 import Pangraph
 
--- * A module containing some exclusivly Hexml helper functions and some which have common interfacing functions.
 
--- | Find the ['Node'] with the final in the ['ByteString'] after following the
--- 'Node' names recursively.
+-- * A module containing containing extra Hexml functions as well as functions to ease conversions between the libraries.
+
+type HexmlAttribute = Text.XML.Hexml.Attribute
+type PangraphAttribute = Pangraph.Attribute
+
+-- | Find the ['Node'] with the final name in the ['ByteString'] after following the 'Node' names recursively, including the root node's name.
+-- Given `[]` returns '[]'.
 followChildren :: Node -> [ByteString] -> [Node]
-followChildren h [] = [h]
-followChildren h bs = (concatMap recurse . childrenBy h) (head bs)
+followChildren _ [] = []
+followChildren h bs = if head bs == name h
+    then concatMap (`recurse` tail bs) $ childrenBy h (bs !! 1)
+    else error $ "Followchildren failed (Did you include the root name?):\n" ++ show h ++ "\n Bytestring: \n" ++ show bs
     where
-        recurse :: Node -> [Node]
-        recurse n = followChildren n (tail bs)
+        recurse :: Node -> [ByteString] -> [Node]
+        recurse _ [] = error "Recurse called on empty Bytestring"
+        recurse h' [_] = [h']
+        recurse h' bs' = concatMap (`recurse` tail bs') $ childrenBy h' (bs' !! 1)
 
--- An unsafe version of the 'Text.XML.Hexml.parse' upon failure throws error to stderr.
+-- An unsafe version of the 'Text.XML.Hexml.parse' upon failure throws error to stderr, showing the resulting error message.
 hexmlParse :: ByteString -> Node
```

Comment:
I think we use `unsafe` prefix for unsafe functions -- let's keep doing this for consistency.